pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -735,6 +756,9 @@ func (s *DiscoveryServer) removeCon(conID string) {
 	}
 
 	xdsClients.Record(float64(len(s.adsClients)))
+	if s.Env.StatusReporter != nil {
+		go s.Env.StatusReporter.RegisterDisconnect(conID, []string{ClusterType, ListenerType, RouteType, EndpointType})
```

Comment:
what is the significance of type here? How will we need to change this when we send v3 APIs or https://github.com/istio/istio/pull/23159?

Reply:
I'm not sure I understand what that pull request is changing in terms of what we send to Envoy.  This PR tracks what Envoy has acknowledged, which today includes Routes, Clusters, Listeners, and Endpoints.  Because each subscription is handled and ACK'ed separately, each has to have its own tracking for status.  If we begin sending more types to Envoy, we will need to add them here, assuming each type represents its own subscription...